core/src/androidTest/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/PlaybackResumerTest.kt

```diff
@@ -0,0 +1,61 @@
+package com.pierfrancescosoffritti.androidyoutubeplayer.core
+
+import androidx.test.filters.LargeTest
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.PlayerConstants
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.YouTubePlayer
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.utils.PlaybackResumer
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.utils.loadOrCueVideo
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.verify
+import kotlin.random.Random
+
+/**
+ * This class tests every single player state combination that may occur then calls resume() to
+ * checks if the appropriate youtubePlayer method is called
+ *
+ * Note: The state combinations tested are:
+ *  lifecycle Resumed/Paused -> All PlayerErrors -> All PlayerState -> Random second -> Random videoId
+ */
+@LargeTest
+class PlaybackResumerTest {
+
+    private val isLifeCycleResumedStates = arrayOf(true, false)
+
+    @Test
+    fun testResumeCallsAppropriateYouTubePlayerMethodUsingVariousStates() {
```

Comment:
There is way too much going on in this tests, I struggle to understand what's happening. I'd suggest splitting it into multiple tests.

In general, when testing, is good practice to avoid writing tests that are too generic. It's better to write multiple tests and test every case "manually" instead of writing loops that test everything etc.